features/measurements/impl/src/test/java/org/opennms/netmgt/measurements/impl/JEXLExpressionEngineEnhancedTest.java

```diff
@@ -0,0 +1,340 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2010-2015 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2015 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.measurements.impl;
+
+import static org.junit.Assert.*;
+
+import java.util.Map;
+
+import org.junit.Test;
+import org.opennms.netmgt.measurements.api.ExpressionEngine;
+import org.opennms.netmgt.measurements.api.exceptions.ExpressionException;
+import org.opennms.netmgt.measurements.api.FetchResults;
+import org.opennms.netmgt.measurements.model.Expression;
+import org.opennms.netmgt.measurements.model.QueryRequest;
+import org.opennms.netmgt.measurements.model.Source;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+public class JEXLExpressionEngineEnhancedTest {
+	private static final double DELTA = 1e-15;
+
+	private final ExpressionEngine jexlExpressionEngine = new JEXLExpressionEngine();
+
+	@Test
+	public void checkSamplefn5NaN() {
+		final String expression = "fn:arrayNaN(\"x\", 5)";
+
+		StringBuffer sb = new StringBuffer("");
+		try{
+
+			double[] result = performExpression(expression);
+
+			for (double value: result){
+				sb.append(value+",");
+			}
+			System.out.println("JEXLExpressionEngineEnhancedTest: expression="+expression +"\n    result "+sb.toString());
+			
+			assertEquals(Double.NaN, result[0], DELTA);
+			assertEquals(Double.NaN, result[1], DELTA);
+			assertEquals(Double.NaN, result[2], DELTA);
+			assertEquals(Double.NaN, result[3], DELTA);
+			assertEquals(Double.NaN, result[4], DELTA);
+			assertEquals(5, result[5], DELTA);
+			assertEquals(6, result[6], DELTA);
+
+		} catch (Exception e){
```

Comment:
Another note: There's no need for these catch blocks, throwing the exceptions directly at the method level would have the same effect.